### PR TITLE
Implement Google Cloud Pub/Sub connectors

### PR DESCRIPTION
**Pull request by IyeOnline**
This PR implements connectors for subscribing/publishing to google cloud subscriptions/topics.

For now the connector does not support any URI-style usage as a review of the interactions between connectors, parsers and the planned TQL2 `from` and `to` operators is planned very soon.

The PR also moves the S3 connector from a builtin to a plugin, so that it can be disabled. This is necessary as it sometimes causes a segmentation fault on initialization  when running with address sanitizer.

Fixes https://github.com/tenzir/issues/issues/2228

### CODE REVIEW

File: plugins/google_cloud_pubsub/src/plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/location.hpp>
+#include <tenzir/plugin.hpp>
+
+#include <arrow/filesystem/azurefs.h>
```
Comment (mavam) [inline plugins/google_cloud_pubsub/src/plugin.cpp:13]: Needed?

File: plugins/google-cloud-pubsub/include/saver.hpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+namespace pubsub = ::google::cloud::pubsub;
+
+class saver final : public plugin_saver {
+public:
+  struct args {
+    located<std::string> project_id;
+    located<std::string> topic_id;
+
+    auto add_to(argument_parser& parser) -> void {
```
Comment (tobim) [inline plugins/google-cloud-pubsub/include/saver.hpp:31]: This is a header file with several non-inline function definitions. I think you should either move them out or make them inline.

**Review by tobim — APPROVED** [review]
You still need to move a couple of definitions out of the `loader.hpp` and `saver.hpp` headers, but otherwise I'm fine with this. I tested the happy and some of the sad paths.

File: plugins/google-cloud-pubsub/include/saver.hpp (hunk 2 above)
Comment (IyeOnline) [reply plugins/google-cloud-pubsub/include/saver.hpp:31]: They are all defined in-class, so they are implicitly inline definitions. I can add an explicit `inline` specifier though.

**Pull request merged**